pkg/controller/kogitosupportingservice/kogitosupportingservice_controller.go

```diff
@@ -0,0 +1,239 @@
+	// Fetch the KogitoSupportingService instance
+	instance, resultErr := fetchKogitoSupportingService(r.client, request.Name, request.Namespace)
+	if resultErr != nil {
+		return
+	}
+
+	// Do not trigger the Reconcile of kogitoSupportingController for ignored services
+	if contains(ignoreServices, instance.Spec.ServiceType) {
+		return
+	}
+	log.Infof("Reconciling KogitoSupportingService for %s in %s", request.Name, request.Namespace)
+	log.Debugf("Going to reconcile service of type %s", instance.Spec.ServiceType)
+	if resultErr = ensureSingletonService(r.client, request.Namespace, instance.Spec.ServiceType); resultErr != nil {
+		return
+	}
+
+	supportingResource := getKogitoSupportingResource(instance)
+
+	requeueAfter, resultErr := supportingResource.Reconcile(r.client, instance, r.scheme)
+	if resultErr != nil {
+		return
+	}
+
+	if requeueAfter > 0 {
+		log.Infof("Waiting for all resources to be created, scheduling for 30 seconds from now")
+		result.RequeueAfter = requeueAfter
+		result.Requeue = true
+	}
+	return
+}
+
+func fetchKogitoSupportingService(client *client.Client, name string, namespace string) (*v1alpha1.KogitoSupportingService, error) {
+	log.Debugf("going to fetch deployed kogito supporting service instance %s in namespace %s", name, namespace)
+	instance := &v1alpha1.KogitoSupportingService{}
+	if exists, resultErr := kubernetes.ResourceC(client).FetchWithKey(types.NamespacedName{Name: name, Namespace: namespace}, instance); resultErr != nil {
+		log.Errorf("Error occurs while fetching deployed kogito supporting service instance %s", name)
+		return nil, resultErr
+	} else if !exists {
+		return nil, fmt.Errorf("kogito supporting service resource with name %s not found in namespace %s", name, namespace)
+	} else {
+		log.Debugf("Successfully fetch deployed kogito supporting reference %s", name)
+		return instance, nil
+	}
+}
+
+// fetches all the supported services managed by kogitoSupportingService controller
+func getKogitoSupportingResource(instance *v1alpha1.KogitoSupportingService) SupportingServiceResource {
+	log.Debugf("going to fetch related kogito infra resource for given infra instance : %s", instance.Name)
+	switch instance.Spec.ServiceType {
+	case v1alpha1.JobsService:
+		log.Debugf("Kogito Supporting Service reference is for Jobs Service")
+		return &JobsServiceSupportingServiceResource{}
+	case v1alpha1.MgmtConsole:
+		log.Debugf("Kogito Supporting Service reference is for Management Console")
+		return &MgmtConsoleSupportingServiceResource{}
+	case v1alpha1.Explainablity:
+		log.Debugf("Kogito Supporting Service reference is for Explainability Service")
+		return &ExplainabilitySupportingServiceResource{}
+	case v1alpha1.TrustyAI:
+		log.Debugf("Kogito Supporting Service reference is for TrustyAI")
+		return &TrustyAISupportingServiceResource{}
+	case v1alpha1.TrustyUI:
+		log.Debugf("Kogito Supporting Service reference is for TrustyUI")
+	}
```

Comment:
Instead of using this switch case, I believe a map would be cleaner, like:

```go
map[ServiceType]ServiceResource = {
}

// then you can just do a map get:
return mymap[instance.Spec.ServiceType]
```